### PR TITLE
Connect team event frontend and backend to submit member form

### PR DESCRIPTION
**Pull request by riyajaggi**
### Summary 

This pull request is the first step towards connecting the frontend to backend for Team Event Form. This PR takes care of the member side of the form.

- [x] Add TeamEventAPI for frontend
- [x] Fetch team events from firebase collection (and remove hard coded examples)
- [x] Add user to pending requests when they submit team event form 
- [ ] TODO: save uploaded image

**Form submitted by member**
image

**Updated team event collection with submission**
image

### Test Plan 

need to test as a non-admin/user

### Breaking Changes 

- Updated the datatypes for the team events to connect frontend/backend (would like some feedback on this)

### CODE REVIEW

File: frontend/src/API/TeamEventsAPI.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+export type TeamEvent = {
+  name: string;
+  date: string;
+  numCredits: string;
+  hasHours: boolean;
+  requests: TeamEventAttendance[];
+  attendees: TeamEventAttendance[];
+  uuid: string;
+};
```
Comment (henry-li-06) [inline frontend/src/API/TeamEventsAPI.ts:24]: Isn't this in `common-types/index.d.ts` already?

Comment (henry-li-06) [inline frontend/src/API/TeamEventsAPI.ts:24]: Aren't these in `common-types/index.d.ts` already?

File: backend/src/DataTypes.d.ts [hunk 2]
```diff
@@ -42,22 +42,34 @@ export type SignInForm = {
...
+
 export type DBTeamEvent = {
   name: string;
   date: string;
   numCredits: string;
   hasHours: boolean;
-  requests: firestore.DocumentReference[];
-  attendees: firestore.DocumentReference[];
+  requests: DBTeamEventAttendance[];
+  attendees: DBTeamEventAttendance[];
```
Comment (henry-li-06) [inline backend/src/DataTypes.d.ts:57]: I think these look good, they should be the same changes as in #240.

File: frontend/src/API/TeamEventsAPI.ts (hunk 1 above)
Comment (riyajaggi) [reply frontend/src/API/TeamEventsAPI.ts]: yes, fixed!

File: backend/src/team-eventsAPI.ts [hunk 3]
```diff
@@ -3,11 +3,10 @@ import { TeamEvent } from './DataTypes';
...
 import PermissionsManager from './permissions';
 
-export const getAllTeamEvents = async (user: IdolMember): Promise<TeamEvent[]> => {
-  const canCreateTeamEvent = await PermissionsManager.canEditTeamEvent(user);
-  if (!canCreateTeamEvent) throw new PermissionError('does not have permissions');
-  return TeamEventsDao.getAllTeamEvents();
-};
+export const getAllTeamEvents = (user: IdolMember): Promise<TeamEvent[]> =>
+  // const canCreateTeamEvent = await PermissionsManager.canEditTeamEvent(user);
+  // if (!canCreateTeamEvent) throw new PermissionError('does not have permissions');
```
Comment (airaagrawal1) [inline backend/src/team-eventsAPI.ts:8]: I think some TeamEventAPI tests should work now, so maybe comment these back in?

Comment (riyajaggi) [reply backend/src/team-eventsAPI.ts:8]: done! thanks for reminding me!

**Comment by riyajaggi** [comment]
Tested as a non-admin: form submits and firebase collection is updated.

**Review by henry-li-06 — APPROVED** [review]

**Pull request merged**